worlds/sc2/client.py

```diff
@@ -519,6 +521,29 @@ def color_code(self, code: str) -> str:
         return '<c val="' + self.color_codes[code] + '">'
 
 
+class SC2JSONtoKivyParser(KivyJSONtoTextParser):
+    def _handle_item_name(self, node: JSONMessagePart):
+        item_name = node["text"]
+        if not item_name in item_descriptions:
+            return super()._handle_item_name(node)
+        
+        flags = node.get("flags", 0)
+        item_types = []
+        if flags & 0b001:  # advancement
+            item_types.append("progression")
+        if flags & 0b010:  # useful
+            item_types.append("useful")
+        if flags & 0b100:  # trap
+            item_types.append("trap")
+        if not item_types:
+            item_types.append("normal")
+
+        # TODO: Some descriptions are too long and get cut off. Is there a general solution or does someone need to manually check every description?
+        ref = "Item Class: " + ", ".join(item_types) + "<br><br>" + item_descriptions[item_name].replace("\n", "<br>")
```

Comment:
As I wrote on the item descriptions, we can do an automated solution of giving every sentence its own line. So just add a `.replace('. ', '.<br>')`.

Another more general option is implementing line-wrap, by splitting on spaces and inserting a `<br>` whenever the next word added would take us over a threshold.

It's also cool to leave this as-is for now and just see how irritating it is / let ideas stew.

Reply:
I've gone with the former, I don't feel great about trying to make a sensible line-wrap with a variable-width font, especially when the window size is not constant.

This does end up with edge cases like the below, but it's better than the alternative.
![python_q9JKOzPARA](https://github.com/user-attachments/assets/d4edcd37-b476-4519-ae69-38d9127c5bd7)

Reply:
I think that example is entirely reasonable. I'm more worried about the cases where we have very long sentences that don't get split up; at least the player has easy recourse in just expanding the window. Simple code is also easier to revisit later, so I think this is the best route for now.